clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -100,19 +100,15 @@ public static List<InetSocketAddress> parseAndValidateAddresses(List<String> url
             }
         }
         if (addresses.isEmpty())
-            dedupeAndHandleMessage("No resolvable bootstrap server in provided urls: " + String.join(",", urls), true);
+            throw new ConfigException("No resolvable bootstrap server in provided urls: " + String.join(",", urls));
```

Comment:
there are other issues that leads to ConfigException and are non-retriable. 
If we want to retry on this error, we probably should look for a exception that we know can be safely retried.